src/Plugin/media/Source/Embeddable.php

```diff
@@ -0,0 +1,155 @@
+ *   allowed_field_types = {"string", "string_long"},
+ * )
+ */
+class Embeddable extends OEmbed {
+
+  /**
+   * Gets the value for a metadata attribute for a given media item.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   * @param string $attribute_name
+   *   Name of the attribute to fetch.
+   *
+   * @return mixed|null
+   *   Metadata attribute value or NULL if unavailable.
+   */
+  public function getMetadata(MediaInterface $media, $name) {
+    if ($this->hasUnstructured($media)) {
+      return $this->getUnstructuredMetadata($media, $name);
+    }
+    return parent::getMetadata($media, $name);
+  }
+
+  /**
+   * Gets the value for a metadata attribute for a given media item.
+   * This is an alternate version to account for unstructured embeds.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   * @param string $attribute_name
+   *   Name of the attribute to fetch.
+   *
+   * @return mixed|null
+   *   Metadata attribute value or NULL if unavailable.
+   */
+  public function getUnstructuredMetadata(MediaInterface $media, $name) {
+    switch ($name) {
+      case 'default_name':
+        if ($title = $this->getMetadata($media, 'title')) {
+          return $title;
+        }
+        elseif ($url = $this->getMetadata($media, 'url')) {
+          return $url;
+        }
+        return parent::getMetadata($media, 'default_name');
+
+      case 'thumbnail_uri':
+        parent::getMetadata($media, 'thumbnail_uri');
+      case 'type':
+      case 'title':
+      case 'author_name':
+      case 'author_url':
+      case 'provider_name':
+      case 'provider_url':
+      case 'cache_age':
+      case 'thumbnail_width':
+      case 'thumbnail_height':
+      case 'url':
+      case 'width':
+      case 'height':
+        return null;
+
+      case 'html':
+        return $media->get('field_media_embeddable_code')->getValue();
+
+      default:
+        break;
+    }
+    return NULL;
+
+  }
+
+
+  /**
+   * Is there a value for the oEmbed URL?
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   *
+   * @return bool
+   *   TRUE means it has an Unstructured embed, FALSE means that field is empty
+   */
+  public function hasOEmbed(MediaInterface $media) {
+    if (!empty($media->get('field_media_embeddable_oembed')->getValue())) {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * Is there a value for the Unstructured Embed?
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   *
+   * @return bool
+   *   TRUE means it has an Unstructured embed, FALSE means that field is empty
+   */
+  public function hasUnstructured(MediaInterface $media) {
+    if (!empty($media->get('field_media_embeddable_code')->getValue())) {
```

Comment:
You can eliminate the `if` statement by just returning the value you are comparing.:
`return !empty($media->get('field_media_embeddable_oembed')->getValue());`

Reply:
fixed!